src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -450,9 +449,11 @@ internal void FunctionCompleted(FunctionStartedEvent startedEvent)
                 RaiseFunctionMetricEvent(startedEvent, _activeFunctionCount, DateTime.UtcNow);
             }
 
-            internal void StopEtwTaskAndRaiseFinishedEvent()
+            internal void StopTimerAndRaiseFinishedEvent()
             {
-                _etwTaskCancellationSource.Cancel();
+                // stop the timer if it has been started
+                _activityTimer?.Change(Timeout.Infinite, Timeout.Infinite);
+
                 RaiseMetricsPerFunctionEvent();
             }
```

Comment:
Is there any interesting race condition here where we're in the middle of `TimerFlush` when we call Stop and then we exit before `TimerFlush` is complete?

This is mostly a brainstorming question -- I don't see any issue but wanted to see if you did.

Reply:
I don't believe so. The TimerFlush will be on a background thread, so will complete. Anyhow, it'll behave as it did before - I've just added better error handling.